### PR TITLE
Retry setup connection until times out (Fixes #3379)

### LINKED ISSUE #3379: helm init --wait gets a context deadline exceeded
```
Running helm init --wait --upgrade --service-account tiller
$HELM_HOME has been configured at /root/.helm.
Error: could not ping Tiller: context deadline exceeded
```
Could be related to using upgrade?

v2.8.0

### PR DESCRIPTION
**Pull request by ghost**
Signed-off-by: Alex Johnson <ajohnson@bombora.com>

### CODE REVIEW

**Comment by TheOriginalAlex** [comment]
@bacongobbler: Could you please review this when you get a sec?

File: cmd/helm/init.go [hunk 1]
```diff
@@ -117,6 +120,7 @@ func newInitCmd(out io.Writer) *cobra.Command {
 	f.BoolVar(&i.dryRun, "dry-run", false, "do not install local or remote")
 	f.BoolVar(&i.skipRefresh, "skip-refresh", false, "do not refresh (download) the local repository cache")
 	f.BoolVar(&i.wait, "wait", false, "block until Tiller is running and ready to receive requests")
+	f.IntVar(&i.timeout, "timeout", 10, "number of seconds to wait for tiller before timing out")
```
Comment (helgi) [inline cmd/helm/init.go]: Is 10 not low? Have you looked at how long it takes it to startup when you have a lot of releases?

Comment (TheOriginalAlex) [reply cmd/helm/init.go]: This is starting tiller.  I thought you start the tiller and then deploy releases.

**Comment by bacongobbler** [comment]
Hey @TheOriginalAlex, can you possibly take a look at #3506? Does that potentially collide with this PR in that it fixes up this gap?

**Comment by bacongobbler** [comment]
ping

**Comment by TheOriginalAlex** [comment]
Ah.  Yep.  I think that's a much better solution than polling like I'm doing here.  Gonna close this PR.

**Pull request closed without merging**